lib/producer/beast.go

```diff
@@ -32,75 +32,80 @@ func (p *producer) beastScanner(scan *bufio.Scanner) error {
 }
 
 // ScanBeast is a splitter for BEAST format messages
-func ScanBeast(data []byte, atEOF bool) (advance int, token []byte, err error) {
-	if atEOF && len(data) == 0 {
-		return 0, nil, nil
-	}
+func ScanBeast() func(data []byte, atEOF bool) (int, []byte, error) {
+	var token [50]byte
```

Comment:
this is our scratch token buffer that we reuse

it is in scope of the returned function, and since it is not shared with any other thing no locks are needed